### ISSUE TITLE
Cannot share image on Android API level 19

### ISSUE THREAD

**Issue by tjvantoll** [tjvantoll/nativescript-social-share#2]
http://nsbuild01.telerik.com:8080/build/view/VerifiedPlugins/job/tns-plugins-tests-socialshare-release-android-api19-emulator/8/artifact/target/surefire-reports/logs/consoleLog.txt

dimitartopuzov [1:23 AM]
search for "File: "/data/data/com.tjvantoll.demo/files/app/tns_modules/nativescript-social-share/social-share.js, line: 30, column: 19"

**Comment by tjvantoll** [comment]
After playing with this for several hours I have an order of magnitude more respect for @EddyVerbruggen's plugin. The problem is that `android.provider.MediaStore.Images.Media.insertImage` appears to rely on an SD card being present on the device, and that it's not a reliable way to write a temporary file on Android.

Eddy's plugin does some crazy work to create a directory, write a new image to it, and then retrieve that file using the stupidly verbose Java APIs. I'm slowly coming to terms with the necessity of this approach, but I'm having some trouble getting the exact syntax correct.

Here's where I'm at for `shareImage()`:

```js
numberOfImagesCreated ++;

var intent = getIntent("image/jpeg");

var stream = new java.io.ByteArrayOutputStream();
image.android.compress(android.graphics.Bitmap.CompressFormat.JPEG, 100, stream);

var imageDirectoryPath = context.getExternalFilesDir(null) + "/socialsharing-downloads";
var imageFileName = "TempFile" + numberOfImagesCreated;

var dir = new java.io.File(imageDirectoryPath);
dir.mkdirs();

var fos = new java.io.FileOutputStream(new java.io.File(dir, imageFileName));
fos.write(stream.toByteArray());

fos.flush();
fos.close();

intent.putExtra(android.content.Intent.EXTRA_STREAM,
	android.net.Uri.parse(imageDirectoryPath + "/" + imageFileName));

share(intent, subject);
```

This doesn't throw any errors when it saves the file, but it does when the share widget later tries to retrieve the saved file:

```
E/Mms/media( 1355): IOException caught while opening or reading stream
E/Mms/media( 1355): java.io.FileNotFoundException: No content provider: /storage/emulated/0/Android/data/com.tjvantoll.demo/files/socialsharing-downloads/TempFile1
```

So I'm guessing I'm saving garbage rather than the correct bytes for the file, but I'm having trouble debugging the problem. I'll dig into this later but I have to step away from this for now.

**Comment by dtopuzov** [comment]
Same crash is reproduced on Android 6 (Api Level 23)

**Comment by tjvantoll** [comment]
I addressed older Android with 34b16a8eb54b146d8bb74a463933261bcf95d6e0, so image sharing should now work in Android 4.2–5.1. However, the Android 6 problem is different, and [also present in Eddy’s Cordova plugin](https://github.com/EddyVerbruggen/SocialSharing-PhoneGap-Plugin/issues/438) so I’m going to open up a new issue to cover that.

**Comment by tjvantoll** [comment]
The new issue here https://github.com/tjvantoll/nativescript-social-share/issues/3.
